session/null.c

```diff
@@ -0,0 +1,63 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <wayland-server.h>
+#include <wayland-util.h>
+#include <wlr/session/interface.h>
+#include <wlr/util/log.h>
+
+const struct session_impl session_null;
+
+static bool null_session_change_vt(struct wlr_session *session, unsigned vt) {
+	return false;
+}
+
+static void null_session_close(struct wlr_session *session, int fd) {
+	// Calling this function is a programming error
+	wlr_log(L_ERROR, "Invalid call to wlr_session_close with null session");
+	exit(1);
+}
+
+static int null_session_open(struct wlr_session *session, const char *path) {
+	// Calling this function is a programming error
+	wlr_log(L_ERROR, "Invalid call to wlr_session_open with null session");
+	exit(1);
+}
+
+static void null_session_finish(struct wlr_session *session) {
+	free(session);
+}
+
+static struct wlr_session *null_session_start(struct wl_display *disp) {
+	if (!getenv("WAYLAND_DISPLAY") && !getenv("_WAYLAND_DISPLAY") && !getenv("DISPLAY")) {
```

Comment:
Yeah, it is pretty crappy. I cannot think of a reliable way to test "is going to be drm", though.